lib/matplotlib/collections.py

```diff
@@ -345,6 +345,14 @@ def draw(self, renderer):
                     and extents.height < self.figure.bbox.height):
                 do_single_path_optimization = True
 
+        # The single path optimization code path is currently broken when
+        # offsets are specified
+        # (https://github.com/matplotlib/matplotlib/issues/16496)
+        # This should be fixed eventually, but for now don't use the optimized
+        # code path if some of the offsets are non-zero
+        if not np.all(offsets == 0) and self._offset_position == 'data':
```

Comment:
```suggestion
        if np.any(offsets != 0) and self._offset_position == 'data':
```
I would find this slightly more readable but won't block over this.